File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        int n = nums.length;
        int totalSubsets = 1 << n; //2^n subsets
        for (int mask = 0; mask < totalSubsets; mask++) {
            List<Integer> list = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                if ((mask & (1 << i)) != 0) {
                    list.add(nums[i]);
                }
            }
            ans.add(list);
        }
        return ans;
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        n = len(nums)
        total_subsets = 1 << n  # 2^n
        
        for mask in range(total_subsets):
            subset = []
            for i in range(n):
                if (mask & (1 << i)) != 0:
                    subset.append(nums[i])
            ans.append(subset)
            
        return ans